### PR TITLE
Add YANG model for FABRIC_PORT

### PR DESCRIPTION
**Pull request by jfeng-arista**
#### Why I did it
YANG model for FABRIC in CONFIG_DB was added in https://github.com/sonic-net/sonic-buildimage/pull/14170.

Need to wait https://github.com/sonic-net/sonic-buildimage/pull/14170 get merged first. 
#### How I did it
Added new YANG model and associated unit tests.
#### How to verify it
Passing unit tests
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf isolateStatus {
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-port.yang:30]: please add description of the leaf nodes

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    }
+                }
+
+                leaf alias {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf lanes {
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-port.yang:42]: I think, the fabric port configuration without specifying `lanes` is not valid, can we add this field as `mandatory`

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf isolateStatus {
+                    type string {
```
Comment (arlakshm) [inline src/sonic-yang-models/yang-models/sonic-fabric-port.yang:31]: if the `isolateStatus` is not speicified should there be a   default value?

**Review by arlakshm — COMMENTED** [review]
as comments

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang (hunk 2 above)
Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-port.yang:42]: fixed.

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang (hunk 3 above)
Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-port.yang:31]: Added

File: src/sonic-yang-models/yang-models/sonic-fabric-port.yang (hunk 1 above)
Comment (jfeng-arista) [reply src/sonic-yang-models/yang-models/sonic-fabric-port.yang:30]: Added

**Review by arlakshm — DISMISSED** [review]

**Comment by qiluo-msft** [comment]
@jfeng-arista Could you resolve conflict?

**Comment by jfeng-arista** [comment]
> @jfeng-arista Could you resolve conflict?

sure

**Comment by kenneth-arista** [comment]
@arlakshm can we merge this PR before the 202305 cut off?

**Comment by arlakshm** [comment]
Hi @jfeng-arista, can you fix the conflict

**Comment by gechiang** [comment]
@jfeng-arista please resolve the conflict so we can try to merge it.

**Comment by kenneth-arista** [comment]
Jie's on FTA for another week. I fixed the conflict locally, but don't have permission to update her repo.

**Comment by gechiang** [comment]
> Jie's on FTA for another week. I fixed the conflict locally, but don't have permission to update her repo.

@kenneth-arista Thanks for responding.  If we choose to wait for @jfeng-arista to address this it would take at least another week.
My suggestion is why not for you to raise a new PR which include everything to replace this PR? This might be faster to get this done.

**Comment by kenneth-arista** [comment]
Thanks for the suggestion @gechiang . I created the clone PR resolving the conflicts here: https://github.com/sonic-net/sonic-buildimage/pull/15629

**Comment by arlakshm** [comment]
closing this PR to avoid confusion

**Pull request closed without merging**